### PR TITLE
Allow CMake build type to be case-insensitive

### LINKED ISSUE #14238: cmake build type is needlessly case sensitive
Drake is the only library I have seen do this: if the build type is not "Debug", "Release", "RelWithDebInfo", etc., the build fails, e.g.:

```
drum @ caesar ~/drake/build
└─ $ ▶ cmake .. -DCMAKE_BUILD_TYPE=debug
CMake Warning at CMakeLists.txt:190 (message):
  Configuration debug is NOT supported


-- Configuring done
-- Generating done
-- Build files have been written to: /home/drum/drake/build
```

### PR DESCRIPTION
**Pull request by BetsyMcPhail**
Fixes #14238

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16283)

### CODE REVIEW

**Comment by BetsyMcPhail** [comment]
@drake-jenkins-bot linux-bionic-clang-cmake-experimental-debug please.

**Comment by jwnimmer-tri** [comment]
@BetsyMcPhail should we assign +@mwoehlke-kitware feature-review this?

File: CMakeLists.txt [hunk 1]
```diff
@@ -191,7 +191,15 @@ else()
...
 
-  if(NOT CMAKE_BUILD_TYPE IN_LIST SUPPORTED_BUILD_TYPES)
+  # To allow the build type to be case-insensitive,
+  # use this variable for any string comparisions
+  string(TOLOWER "${CMAKE_BUILD_TYPE}" BUILD_TYPE)
+
+  string(TOLOWER "${SUPPORTED_BUILD_TYPES_STRING}" BUILD_TYPES_STRING_LOWER)
+  string(REPLACE " " ";" BUILD_TYPES_LOWER
+    "${BUILD_TYPES_STRING_LOWER}"
+  )
```
Comment (mwoehlke-kitware) [inline CMakeLists.txt:201]:
```suggestion
  string(TOLOWER "${SUPPORTED_BUILD_TYPES}" BUILD_TYPES_LOWER)
```

**Review by mwoehlke-kitware — COMMENTED** [review]
`BUILD_TYPE` vs. `BUILD_TYPES_LOWER` vs. `SUPPORTED_BUILD_TYPES` seems a bit inconsistent; not sure I have a better idea, though.

**Review by BetsyMcPhail — COMMENTED** [review]
I changed to `BUILD_TYPE_LOWER`, I don't know if that's really much better. `SUPPORTED_BUILD_TYPES_LOWER` seemed to be getting a little long...  

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16283#-:-Mt4wU6l31JG4iYuCPGK:bcqy56d)* status: 1 unresolved discussion, LGTM missing from assignee mwoehlke-kitware, needs platform reviewer assigned, needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits), missing label for release notes (waiting on @mwoehlke-kitware)

___
*[CMakeLists.txt, line 201 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16283#-MrhAqlEg1YbmnOIJATU-r1-201:-Mt4wEeb5m4d0W5AXVLM:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/434a9cdfc62a3462cfe4b7ce5b0fa2e542bbbf6a/CMakeLists.txt#L201)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

```suggestion
  string(TOLOWER "${SUPPORTED_BUILD_TYPES}" BUILD_TYPES_LOWER)
```
</blockquote></details>

Done.

**Review by mwoehlke-kitware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16283)* status: 1 unresolved discussion, LGTM missing from assignee mwoehlke-kitware, needs platform reviewer assigned, needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits), missing label for release notes (waiting on @BetsyMcPhail and @mwoehlke-kitware)

___
*[CMakeLists.txt, line 196 at r2](https://reviewable.io/reviews/robotlocomotion/drake/16283#-Mt9WUhx11PnvQsqEoVe:-Mt9WUhx11PnvQsqEoVf:bk50sad) ([raw file](https://github.com/robotlocomotion/drake/blob/a0b3a497cbf9bb585e99f4412b08bc5f230ae2e0/CMakeLists.txt#L196)):*
> ```CMake
>   # To allow the build type to be case-insensitive,
>   # use this variable for any string comparisions
>   string(TOLOWER "${CMAKE_BUILD_TYPE}" BUILD_TYPE_LOWER)
> ```

:+1:, I think renaming this helps a lot with readability :slightly_smiling_face:.
___
*[CMakeLists.txt, line 198 at r2](https://reviewable.io/reviews/robotlocomotion/drake/16283#-Mt9W6Yh6uTbt7PiVPyo:-Mt9W6Yh6uTbt7PiVPyp:b-mwg05g) ([raw file](https://github.com/robotlocomotion/drake/blob/a0b3a497cbf9bb585e99f4412b08bc5f230ae2e0/CMakeLists.txt#L198)):*
> ```CMake
>   string(TOLOWER "${CMAKE_BUILD_TYPE}" BUILD_TYPE_LOWER)
> 
>   string(TOLOWER "${SUPPORTED_BUILD_TYPES}" BUILD_TYPES_LOWER)
> ```

Given this is the only use, `SUPPORTED_BUILD_TYPES_LOWER` might not be too bad? Yes, it's a bit clunky, but it only appears here.

**Review by mwoehlke-kitware — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16283#-:-MtIy5e-3urdtX02K1NR:bnfp4nl)* status: needs platform reviewer assigned, needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits), missing label for release notes (waiting on @mwoehlke-kitware)

**Review by mwoehlke-kitware — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16283)* status: needs platform reviewer assigned, needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits), missing label for release notes (waiting on @BetsyMcPhail)

**Review by BetsyMcPhail — COMMENTED** [review]
+@ericcousineau-tri for platform review

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16283#-:-MtJ1kOQDt6IlUrAWJvz:b-6jso97)* status: LGTM missing from assignee EricCousineau-TRI(platform), commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @EricCousineau-TRI)

**Review by EricCousineau-TRI — COMMENTED** [review]
:lgtm: platform

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16283#-:-MtJ6SHxEgi3uOQV8Ghc:bcpz082)* status: commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @BetsyMcPhail)

**Pull request merged**